apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java

```diff
@@ -233,9 +236,19 @@ private void assertMetaDataReady() throws UninitializedException {
      */
     @Override
     public void blockUntilReady() throws Exception {
-        if (serializedMetaData == null) {
+
```

Comment:
This feels a bit racy to me, as `blockUntilReady` is called from multiple Threads I think and at the same time reads/writes multiple fields.

What about just making `blockUntilReady()` synchronized?
If it is not ready, it blocks anyway. If the metadata is ready, the check is so fast that the `synchronized` should not matter.

This way you can also turn the field `serializedActivationMethod` into a plain `boolean`.

Reply:
Agreed, it's simpler to have a small `synchronized` method here, should be fixed in https://github.com/elastic/apm-agent-java/pull/3095/commits/39b037e28d24cac798c1821719180b256e2a54cb